src/Build/BackEnd/Node/OutOfProcServerNode.cs

```diff
@@ -269,11 +273,40 @@ private void HandlePacket(INodePacket packet)
             switch (packet.Type)
             {
                 case NodePacketType.ServerNodeBuildCommand:
-                    HandleServerNodeBuildCommand((ServerNodeBuildCommand)packet);
+                    HandleServerNodeBuildCommandAsync((ServerNodeBuildCommand)packet);
+                    break;
+                case NodePacketType.ServerNodeBuildCancel:
+                    HandleServerNodeBuildCancel((ServerNodeBuildCancel)packet);
                     break;
             }
         }
 
+        private void HandleServerNodeBuildCancel(ServerNodeBuildCancel command)
+        {
+            BuildManager.DefaultBuildManager.CancelAllSubmissions();
```

Comment:
I wonder, why can't we reuse here the cancellation from the MSBuildApp.Execute() function, by throwing the cancellation event manually? Also, how the things about the server after the current cancellation implementation works? Does server send the ServerNodeBuildResult package with the relevant information? A unit test would be helpful here, as we would see the intended output after the cancellation and the fact that it was finalized.

Reply:
We discussed with Roman what we can reuse from the cancellation code in XMake (Console_CancelKeyPress). It contains some handling for interactive console, but server node works in non-interactive console.

Reply:
I planned to add test when Nathan's PR with server node tests is merged. We can merge his PR first and I will add test for cancelation.